adserver/decisionengine/backends.py

```diff
@@ -393,13 +393,35 @@ def select_flight(self):
 
         return None
 
+    def get_ad_ctr_weight(self, ad):
+        """Apply the ad weighting factor based on the sampled CTR."""
```

Comment:
Seems we should define what the logic of this function is here, similar to in the PR description? It's not immediately obvious what it does:

Reply:
```suggestion
        """Apply the ad weighting factor based on the sampled CTR.

             Give better performing ads slightly more weighting.
             The scale ramps up based on how far different the CTRs are.
             For example, in a flight with 2 ads, ad X with 0.1% CTR and ad Y with 0.2% CTR, 
             the chances for X are 6/16 ~= 37.5% and the chances for Y are 10/16 ~= 62.5%. 
              
             We can play a bit with these weights if we want.
        """
```

Reply:
Almost certainly needs formatting help :)

Reply:
I added the comment with formatting.